pkgs/servers/tailscale/default.nix

```diff
@@ -2,21 +2,21 @@
 
 buildGoModule rec {
   pname = "tailscale";
-  version = "1.4.2";
-  tagHash = "f40ccb086c4c3f09b3cc67b7c559bd2c5d3cf953"; # from `git rev-parse v1.4.2`
+  version = "1.4.4";
+  tagHash = "f40ccb086c4c3f09b3cc67b7c559bd2c5d3cf953"; # from `git rev-parse v1.4.4`
```

Comment:
The tag hash is wrong, should be 4b0494aa7972d50ce1d2228271e48377a22e4680.

Reply:
keeping the commit hash makes it a bit hard automatizing the updates, what do you think of using `src.rev`
```diff
--- i/pkgs/servers/tailscale/default.nix
+++ w/pkgs/servers/tailscale/default.nix
@@ -25,7 +24,7 @@ buildGoModule rec {
   preBuild = ''
     export buildFlagsArray=(
       -tags="xversion"
-      -ldflags="-X tailscale.com/version.Long=${version} -X tailscale.com/version.Short=${version} -X tailscale.com/version.GitCommit=${tagHash}"
+      -ldflags="-X tailscale.com/version.Long=${version} -X tailscale.com/version.Short=${version} -X tailscale.com/version.GitCommit=${src.rev}"
     )
   '';
```
```
$ result/bin/tailscale version
1.4.4
  tailscale commit: v1.4.4
  go version: go1.15.7
```

Reply:
src.rev isn't a commit ID, is it? Or does fetchFromGitHub expand it to the commit hash?

Reply:
is the revision used, in this case the tag. IMHO it should be enough to reference a commit

Reply:
It should be, but we're specifically setting a git commit hash with that value, so it might mess up some of our downstream tools. Let me see what I can do. If we tolerate it being unset, we can stick with embedding just the version number and rely on the tag being static.

Reply:
Okay, the code treats the commit hash as optional. In that case, we can remove `tagHash` completely, and remove ` -X tailscale.com/version.GitCommit=${tagHash}` from the preBuild command. Then this derivation will be completely automatically updatable for future changes. WDYT?

Reply:
sounds good :)